### PR TITLE
Fyle 1vcx46z : Fixed change the split expense success pop up to success toast message

### PR DESCRIPTION
**Pull request by siddharth0-0**
Screenshot 2021-11-27 170159

image

image

### CODE REVIEW

File: src/app/fyle/split-expense/split-expense.page.ts [hunk 1]
```diff
@@ -325,10 +314,28 @@ export class SplitExpensePage implements OnInit {
               return forkJoin(observables$);
             }),
             tap((res) => {
-              this.showSplitExpenseStatusPopup(true);
+              const message = 'The expense has split successfully';
```
Comment (Dimple16) [inline src/app/fyle/split-expense/split-expense.page.ts:317]: `Your expense was split successfully`

File: src/app/fyle/split-expense/split-expense.page.ts [hunk 2]
```diff
@@ -325,10 +314,28 @@ export class SplitExpensePage implements OnInit {
...
+              this.matSnackBar.openFromComponent(ToastMessageComponent, {
+                ...this.snackbarProperties.setSnackbarProperties('success', { message }),
+                panelClass: ['msb-success-with-camera-icon'],
+              });
+              this.trackingService.showToastMessage({ ToastContent: message });
+              this.router.navigate(['/', 'enterprise', 'my_expenses']);
             }),
             catchError((err) => {
-              this.showSplitExpenseStatusPopup(false);
+              const toastMessageData = {
```
Comment (Dimple16) [inline src/app/fyle/split-expense/split-expense.page.ts:326]: Add a screenshot/recording for the failure case

**Review by Dimple16 — CHANGES_REQUESTED** [review]
few comments

File: src/app/fyle/split-expense/split-expense.page.ts [hunk 3]
```diff
@@ -325,10 +314,28 @@ export class SplitExpensePage implements OnInit {
...
+                message: 'Seems to be a stormy day. You should try this after sometime',
```
Comment (Dimple16) [inline src/app/fyle/split-expense/split-expense.page.ts:327]: `Seems to be a stormy day. Please try again after sometime`

**Comment by Dimple16** [comment]
the error toast should also be above the bottom nav (above the pink camera) it shouldn't hide the bottom sections. Can we fix this @siddharth0-0 ?

**Review by Dimple16 — CHANGES_REQUESTED** [review]
few comments

**Review by Dimple16 — APPROVED** [review]

File: src/app/fyle/split-expense/split-expense.page.ts [hunk 4]
```diff
@@ -325,10 +314,28 @@ export class SplitExpensePage implements OnInit {
...
+                message: 'We were unable to split your expense. Please try again later',
+                redirectionText: 'Retry',
+              };
+              const splitExpenseErroeSnackBar = this.matSnackBar.openFromComponent(ToastMessageComponent, {
+                ...this.snackbarProperties.setSnackbarProperties('failure', toastMessageData),
+                panelClass: ['msb-failure-with-camera-icon'],
+              });
+              this.trackingService.showToastMessage({ ToastContent: toastMessageData.message });
+              splitExpenseErroeSnackBar.onAction().subscribe(() => {
+                this.router.navigate(['/', 'enterprise', 'my_expenses']);
```
Comment (Julias0) [inline src/app/fyle/split-expense/split-expense.page.ts:336]: Shouldnt this open up the split page again instead of opening my expenses?

**Review by Dimple16 — APPROVED** [review]

File: src/app/fyle/split-expense/split-expense.page.ts (hunk 4 above)
Comment (siddharth0-0) [reply src/app/fyle/split-expense/split-expense.page.ts]: Removed the retry option as it was leading to same route as default route `my_expense`

**Review by Julias0 — APPROVED** [review]

**Pull request merged**